webui/elm/package-lock.json

```diff
@@ -673,7 +673,12 @@
       "requires": {
         "acorn-node": "^1.6.1",
         "defined": "^1.0.0",
-        "minimist": "^1.1.1"
+        "minimist": "^1.2.3"
```

Comment:
.

Reply:
> running npm install generates a dirty package-json for me. do you get the same behavior? we should probably check it in if that's the case.

`npm install` will usually generate a new `package-lock.json` which is ok. For CI, the step would be to generate the `package-lock.json` whenever there are changes to `package.json` and then check that in. `npm ci` gets run on the CI so it will prevent from generating a dirty build.

Reply:
> but we're still advertising that we install versions less than 1.2.5 through the package-lock file and I suspect that's what GitHub looks at when determining the security vulnerabilities rather than building our software

The security complaint was only on the elm `package-lock.json` so it was only applied there, but probably makes sense to have it on the react side as well. So if you look at elm's `package-lock.json`, you won't find minimist versions less than `v1.2.3`.

Reply:
> should we add npm-force-resolution to our dev-dependencies?

I was thinking about that as well, but looking at their docs and just general search on this subject I couldn't find any instances of anybody doing this but I don't see why it would be an issue... I'll give it a shot.

Reply:
> > running npm install generates a dirty package-json for me. do you get the same behavior? we should probably check it in if that's the case.
> 
> `npm install` will usually generate a new `package-lock.json` which is ok, there's no guarantees that it will consistently build the same `package-lock.json` file. For CI, the step would be to generate the `package-lock.json` whenever there are changes to `package.json` and then check the resulting `package-lock.json` in. `npm ci` should be the step to run if you just want to get all the dependencies (same as `make get-deps).

you're right that's expected, if those changes don't include minimist updates wouldn't matter if we include it here or not.

Reply:
> The security complaint was only on the elm `package-lock.json` so it was only applied there. I'm auditing via `npm audit` and I suspect that is what github is using as well since it is not complaining about `minimist` for react and only elm where `npm audit` did have an issue for elm. So if you look at elm's `package-lock.json`, you won't find minimist versions less than `v1.2.3`.

right 1.2.3 is the minimum secure version, not 1.2.5 my bad. 
I couldn't find exactly how GitHub determines repositories having security issues for sure but it seems they go based off of [our dependency graph](https://github.com/determined-ai/determined/network/dependencies)  which comes from package-lock.json. What I'm not sure about is why they haven't picked up on react's package-lock.json I assume if/when they do they'll flag any minimist ver below 1.2.3. but that's for later.

> When a new vulnerability is added to the GitHub Advisory Database, we identify public repositories (and private repositories that have opted in to vulnerability detection) that use the affected version of the dependency, send a security alert to repository maintainers, and generate an automated security update.



https://help.github.com/en/github/visualizing-repository-data-with-graphs/listing-the-packages-that-a-repository-depends-on